Move users to a new WordPressCore module

crazytonyli opened the pull request:
~This PR is not buildable yet (pending a wordpress-rs alpha release)~ **Update**: this PR now uses the latest wordpress-rs alpha version. But the changes are fairly simple. I thought we could get the changes reviewed while waiting for the alpha release.

This PR moves the non-UI part of self-hosted user management and some foundational things it uses to a new module `WordPresCore`, which includes strict concurrency checks. The new module will include components communicating with WordPress sites via REST API in WordPress core (hence the name). I'm open to suggestions of better module names.

## Regression Notes
1. Potential unintended areas of impact


2. What I did to test those areas of impact (or what existing automated tests I relied on)


3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

crazytonyli committed:
Move users to a new WordPressCore module

crazytonyli committed:
Adopt wordpress-rs API changes (pending SPM update)

dangermattic commented:
<!--
  0 Errors
  2 Warnings: <code>Modules/Package.swift</c...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>Modules/Package.swift</code> was changed without updating its corresponding <code>Package.resolved</code>. Please resolve the Swift packages as appropriate to your project setup (e.g. in Xcode or by running <code>swift package resolve</code>).</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

kean commented:
> This PR moves the non-UI part of self-hosted user management and some foundational things it uses to a new module WordPresCore, which includes strict concurrency checks. The new module will include components communicating with WordPress sites via REST API in WordPress core (hence the name). I'm open to suggestions of better module names.

Any plans for the Core Data entities and the related services? 

The name seems OK.

jkmassel commented:
> Any plans for the Core Data entities and the related services?

A Core Data-based `DataStore` would be pretty easy to implement – that could be part of this module, or could be implemented elsewhere. This PR is a good start from migrating away from Core Data where it's not needed

crazytonyli committed:
Update wordpress-rs

crazytonyli committed:
Merge branch 'trunk' into move/users-to-module

crazytonyli committed:
Fix unit tests compiling issues

crazytonyli committed:
Update test plan to include the tests in the Modules Swift Package

crazytonyli committed:
Rename WordPressUI's test target as WordPressUIUnitTests

The name WordPressUITests conflicts with the name of WordPress target's
UI Tests target

kean commented on Modules/Package.swift:
```diff
@@ -86,7 +86,7 @@ let package = Package(
         .testTarget(name: "WordPressSharedTests", dependencies: [.target(name: "WordPressShared")], swiftSettings: [.swiftLanguageMode(.v5)]),
         .testTarget(name: "WordPressSharedObjCTests", dependencies: [.target(name: "WordPressShared"), .target(name: "WordPressTesting")], swiftSettings: [.swiftLanguageMode(.v5)]),
-        .testTarget(name: "WordPressUITests", dependencies: [.target(name: "WordPressUI")], swiftSettings: [.swiftLanguageMode(.v5)]),
+        .testTarget(name: "WordPressUIUnitTests", dependencies: [.target(name: "WordPressUI")], swiftSettings: [.swiftLanguageMode(.v5)]),
```
I can see how this was ambiguous.
